### PR TITLE
Add token support (and fix tests)

### PR DESCRIPTION
**Pull request by SmileyChris**
Token authentication support added, along with tests.
Also fixed broken Python 3.2 tests in tox due to recent incompatible upstream test requirements

### CODE REVIEW

**Comment by SmileyChris** [comment]
Looks like the travis config needs updating to use a version of tox also suitable for Python 3.2. I'll leave that outside this pull request.

File: rest_framework_proxy/views.py [hunk 1]
```diff
@@ -82,13 +82,18 @@ def get_headers(self, request):
...
         if username and password:
-            base64string = base64.encodestring('%s:%s' % (username, password)).replace('\n', '')
-            headers['Authorization'] = 'Basic %s' % base64string
+            auth_token = '%s:%s' % (username, password)
+            auth_token = base64.b64encode(auth_token.encode('utf-8')).decode()
+            headers['Authorization'] = 'Basic %s' % auth_token
+        else:
+            auth_token = self.proxy_settings.AUTH.get('token')
+            if auth_token:
+                headers['Authorization'] = 'Token %s' % auth_token
```
Comment (eofs) [inline rest_framework_proxy/views.py]: Looks good but should `Token` string be configurable by user too? The `Basic` is part of the Basic Authentication Scheme specifications but there's also a bunch of other conventions too. For example OAuth usually uses `Bearer` authentication method.

I think more generic solution would be something like this:

```python
if username and password:
   # ... snip ...
else:
  auth_token = self.proxy_settings.AUTH.get('token')
  if auth_token:
    headers['Authorization'] = auth_token
```

This way users may set header to `Token [token]`, `Beare [token]`, `Foo [token]`, `[token]` or any other required value. What do you think?

Comment (SmileyChris) [inline rest_framework_proxy/views.py]: Good point. I'll implement that tomorrow and update the pull request (or
you can just pull and then make the change)

**Comment by SmileyChris** [comment]
Does that look better?

**Comment by eofs** [comment]
Looks good. Thanks!

**Pull request merged**